.github/actions/generate-commit-list-v1/src/doesBranchOrTagExist.ts

```diff
@@ -0,0 +1,31 @@
+import { getExecOutput } from '@actions/exec'
+
+interface DoesBranchOrTagExistParams {
+  branchName?: string
+  tag?: string
+}
+
+/**
+ * Checks if a branch or tag exists in the current git repository.
+ * @param {string} branchName The name of the branch to check.
+ * @param {string} tag The name of the tag to check.
+ * @returns A promise that resolves to a boolean indicating whether the branch or tag exists.
+ */
+
+export default async function doesBranchOrTagExist({
+  branchName,
+  tag
+}: DoesBranchOrTagExistParams): Promise<boolean> {
+  try {
+    const subCommand = tag ? tag : `origin/${branchName}`
+    await getExecOutput(`git rev-parse --verify ${subCommand}`)
```

Comment:
Does this need to account for `stderr` errors?

Reply:
the function `getExecOutput` returns values in stdout if everything is ok and throws an error if not. `stderr` is never thrown. I've tested this